### PR TITLE
Make further improvements to `build.rs`

### PR DESCRIPTION
**Pull request by hannobraun**
Based on #186. I recommend reviewing/merging that one first.

### CODE REVIEW

**Comment by hannobraun** [comment]
Rebased, ready for review.

File: build.rs [hunk 1]
```diff
@@ -25,11 +37,45 @@ fn main() {
     // purpose. `OUT_DIR` points to a directory within target, whose path even
     // contains a hash. This configuration file needs to be referenced from the
     // GDB configuration, which can't just ask Cargo where to look for it.
+    fs::create_dir_all("target")?;
+    File::create("target/openocd.cfg")?.write_all(openocd_cfg)?;
```
Comment (david-sawatzke) [inline build.rs]: If the errors are just bubbled up, why not just `.unwrap()`?

Comment (hannobraun) [reply build.rs]: I think this reads more nicely. `unwrap` and `expect` breaks the flow of what's actually supposed to happen.

In general, I believe that lower layers should report errors, and leave it to upper layers to handle them. But granted, that doesn't make much of a difference in this tiny script.

**Comment by hannobraun** [comment]
(continuing the discussion from #14)

@david-sawatzke I went ahead and implemented full target selection support. I see that you commented in #14 and approved this pull request in the meantime. I still think this was worthwhile work.

Please let me know what you think about the new commits.

**Comment by hannobraun** [comment]
I just saw that the order of commits, as displayed here, is wrong. The first two commits should be switched. The reason for that is, that I regularly rebase while I develop locally, to make the resulting commits easier to read. Unfortunately, GitHub shows commits in pull requests sorted by time, not by actual order. Please rest assured, that the actual order (as shown by `git log`) is correct.

File: memory_32_8.x [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/*
+    Set memory sizes to the smallest values found in any of the  LPC800 parts we
```
Comment (david-sawatzke) [inline memory_32_8.x]: This should be updated, since it's a specific value now (same in the 2 other memory.x files)

**Comment by david-sawatzke** [comment]
We also need to featuregate CAPT & DAC1 now (probably also previously, but now more obviously)

File: memory_32_8.x (hunk 2 above)
Comment (hannobraun) [reply memory_32_8.x]: Right. I've pushed another commit to remove them.

**Comment by hannobraun** [comment]
Merging now, since the previous commit was approved, and the last one only removes some outdated comments.

> We also need to featuregate CAPT & DAC1 now (probably also previously, but now more obviously)

I've opened #191.

**Pull request merged**